### ISSUE TITLE
Crash if path node contains one point

### ISSUE THREAD

**Issue by wnagrodzki** [SVGKit/SVGKit#467]
Parsing the following SVG results in producing CALayer with CGRectNull frame. Attempt to render will crash.

``` svg
<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 18.1.1, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
     viewBox="0 0 1500 1500" enable-background="new 0 0 1500 1500" xml:space="preserve">
     <path fill="#868686" d="M197.6,700.4"/>
</svg>

```

CGRectNull comes from CGPathGetBoundingBox(), this seems like a bug, CGRectZero should be returned instead.

``` swift
/* Return the bounding box of `path'. The bounding box is the smallest
   rectangle completely enclosing all points in the path, including control
   points for Bézier cubic and quadratic curves. If the path is empty, then
   return `CGRectNull'. */

@available(iOS 2.0, *)
public func CGPathGetBoundingBox(path: CGPath?) -> CGRect
```

``` swift
let path = CGPathCreateMutable()
CGPathMoveToPoint(path, nil, 1, 1)
CGPathIsEmpty(path)
>>> false
CGPathGetBoundingBox(path)
>>> CGRectNull
```

Possible fix may be applied in SVGHelperUtilities:342.

``` objective-c
// calculate the rendered dimensions of the path
CGRect r = CGRectInset(CGPathGetBoundingBox(pathRelative), -strokeWidth/2., -strokeWidth/2.);
if (CGRectEqualToRect(r, CGRectNull)) { r = CGRectZero; }
```

There may be a better way to fix this issue, I am not familiar with the whole project.

**Comment by adamgit** [comment]
As far as I can tell, your SVG is invalid - it contains zero points. "Move
to" does not create a point.

Also, I think CGPathIsEmpty does not do what you think it does - it
only asks if the path contains ELEMENTS, not if it contains POINTS.

NB: we cannot replace "null" with "0" - they mean different things.

CGRectNull means there is no rect. A CGRectZero means "a rect at 0,0 which
is very small" - it DOES NOT MEAN "no rect". This is very confusing, but
it's Apple's definition!

The solution is probably to change the rendering method so that if the
bounding box is empty, it cancels rendering that item?

**Comment by wnagrodzki** [comment]
Yes, this auto-generated SVG does not create a single point, but it is still a valid SVG file.

I am aware of the difference between `CGRectNull` and `CGRectZero`.
`CGPathGetBoundingBox()` documentation says "_If the path is empty, then
   return CGRectNull_ ". The path was not empty thus I wouldn't expect CGRectNull there. Documentation is not precise enough.

Yes, canceling rendering seems like a better idea. Maybe such a layer should not be created at all?

**Comment by adamgit** [comment]
The path was empty _of points_, it contained a non-rendering element. As a
path, it was empty; as a CGPath data structure, it wasn't empty. Writing
precise definitions is difficult! :)

"maybe such a layer should not be created at all?" - I think we have to
create the layer because it reflects the structure of the DOM object. But
perhaps size it like a "G" tag - i.e. it is sized after all the sublayers,
and is sized to be the aggregate of all of them?

If there are no sublayers ... I guess we cannot create it, since Apple has
no way of rendering a layer with no contents :(
